app/controllers/sessions_controller.rb

```diff
@@ -10,11 +11,16 @@ def login
       session[:user_id] = response[:user_id]
       redirect_to "/dashboard"
     else
-      flash[:error] = "Invalid credentials"
+      flash[:error] = response[:error]
```

Comment:
Also curious what prompted the change here. just wanting the error message from the json response?